src/hooks/useCategorySelection.tsx

```diff
@@ -0,0 +1,64 @@
+import type { MouseEvent, MouseEventHandler } from 'react';
+import { useReducer } from 'react';
+
+interface IState {
+  categoryList: string[];
+  categoryLength: number;
+}
+
+interface IAction {
+  type: string;
+  category: string;
+}
+
+const initialState: IState = {
+  categoryList: [],
+  categoryLength: 0,
+};
+
+const CLICKED_NUMBER = 5;
+
+const reducer = (state: IState, action: IAction): IState => {
+  switch (action.type) {
+    case 'PUSH':
+      state.categoryLength++;
+
+      if (state.categoryLength > CLICKED_NUMBER) {
+        state.categoryLength = CLICKED_NUMBER;
+      } else {
+        state.categoryList.push(action.category);
+      }
```

Comment:
```suggestion
     if (state.categoryList.length >= 5) return; 
     
     state.categoryList.push(action.category);
```

categoryListLength 가 5 되었을때 early return 해서 아래 로직을 실행하지 않는 방법도 좋을것 같습니다.